pkg/policy/api/rule.go

```diff
@@ -590,3 +648,29 @@ func (kr *PortRuleKafka) GetAPIVersion() (int16, bool) {
 
 	return *kr.apiVersionInt, false
 }
+
+// MapRoleToAPIKey maps the Role to the low level set of APIKeys for that role
+func (kr *PortRuleKafka) MapRoleToAPIKey() error {
+	if len(kr.Role) <= 0 {
+		return fmt.Errorf("Kafka Role is nil")
```

Comment:
Just remove this check, you guard the call to this function with `len(kr.Role) > 0`

Reply:
done